### PR TITLE
Consolidate `humility` subcmds and restore `xtask gdb`

### PR DESCRIPTION
**Pull request by mkeeter**
This consolidates the three Humility-based subcommands (`flash` / `test` / `humility`) to all run through `humility::run`, instead of having separate-but-mostly-overlapping implementations.  It also restores `cargo xtask gdb`, using the same code path.

Drive-by fixes:
- Remove an unneeded `mut`
- Make `flash.ron` pretty-printed

### CODE REVIEW

File: build/xtask/src/main.rs [hunk 1]
```diff
@@ -121,11 +120,28 @@ enum Xtask {
...
+    #[clap(short, long)]
+    verbose: bool,
+
+    /// Extra options to pass to Humility
+    #[clap(last = true)]
+    extra_options: Vec<String>,
+}
+
 // For commands which may execute on specific packages, this enum
 // identifies the set of packages that should be operated upon.
```
Comment (steveklabnik) [inline build/xtask/src/main.rs:138]: I know you didn't modify this line, but I now find it pretty confusing? it is supposedly describing an enum but is describing main?

File: build/xtask/src/main.rs [hunk 2]
```diff
@@ -121,11 +120,28 @@ enum Xtask {
...
 fn main() -> Result<()> {
     let xtask = Xtask::parse();
+    run(xtask)
+}
 
+fn run(xtask: Xtask) -> Result<()> {
```
Comment (steveklabnik) [inline build/xtask/src/main.rs:144]: what's the reason to extract everything but arg parsing into a separate function?

**Review by steveklabnik — APPROVED** [review]
This looks great overall! I have two little non-blocking questions.

File: build/xtask/src/main.rs (hunk 2 above)
Comment (mkeeter) [reply build/xtask/src/main.rs:144]: So I can re-call into `run` [further down](https://github.com/oxidecomputer/hubris/blob/f194910e81ae34e1ac21a07b8a48ddf78697ff78/build/xtask/src/main.rs#L187)

File: build/xtask/src/main.rs (hunk 1 above)
Comment (mkeeter) [reply build/xtask/src/main.rs:138]: lol yeah, it looks like someone (probably me) forgot to delete it when removing `enum RequestedPackages`
```rust
// For commands which may execute on specific packages, this enum
// identifies the set of packages that should be operated upon.
enum RequestedPackages {
    // Specifies a single specific (Package, Target) pair.
    Specific(Option<String>, Option<String>),
    // Specifies the command should operate on all packages.
    All,
}
```

I deleted it now!

**Pull request merged**